src/util-base64.c

```diff
@@ -118,10 +118,10 @@ Base64Ecode DecodeBase64(uint8_t *dest, uint32_t dest_size, const uint8_t *src,
             /* Invalid character found, so decoding fails */
             if (src[i] != '=') {
                 valid = false;
-                if (mode != BASE64_MODE_RELAX) {
+                ecode = BASE64_ECODE_ERR;
+                if (mode == BASE64_MODE_RFC2045) {
```

Comment:
Upon further checking, decoded bytes should only be reset for STRICT mode which is only used by datasets in which case if we are for example unable to decode a base64 encoded url, we should completely ignore that.
Is this a correct assumption @victorjulien w.r.t. using base64 strict mode w datasets?